src/pevm.rs

```diff
@@ -62,9 +64,14 @@ pub fn execute<S: Storage + Send + Sync>(
     let Some(block_env) = get_block_env(&block.header) else {
         return Err(PevmError::MissingHeaderData);
     };
-    let Some(tx_envs) = get_tx_envs(&block.transactions) else {
-        return Err(PevmError::MissingTransactionData);
-    };
+    let tx_envs = block
+        .transactions
+        .as_transactions()
+        .ok_or(PevmError::MissingTransactionData)?
+        .iter()
+        .map(get_tx_env)
+        .collect::<Result<Vec<TxEnv>, TransactionParsingError>>()
+        .map_err(PevmError::InvalidTransaction)?;
 
     // TODO: Continue to fine tune this condition.
     if force_sequential || tx_envs.len() < 4 || block.header.gas_used <= 650_000 {
```

Comment:
I just realized we're only using `block.header.gas_used`  after all the conversions.

Let's copy `gas_used` to a local variable so `get_tx_env` can take ownership of the `Transaction`s and avoid the expensive clones (`tx.input`, `tx.access_list`, implicit byte copies, etc.).

Currently, we're cloning the transaction data just to drop them with `block` at the end of this scope :cry:.

Reply:
I just tried this locally and it turns out we don't need to copy `gas_used`. Rust's smart enough to know the partial move of the block transactions doesn't clash with its header. Nice!